### PR TITLE
provide default implementation for the previous readFilter interface

### PR DESCRIPTION
**Pull request by ArneBab**
This keeps compatibility with plugins using the readFilter.

### CODE REVIEW

File: src/freenet/client/filter/ContentDataFilter.java [hunk 1]
```diff
@@ -35,7 +35,17 @@
...
 	 */
-	public void readFilter(
+	void readFilter(
 			InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
 			String schemeHostAndPort, FilterCallback cb) throws DataFilterException, IOException;
+
+	/**
+	 * Compatibility for readFilter without schemeHostAndPort. Please use readFilter with schemeHostAndPort.
+	 */
+	default void readFilter(
```
Comment (Bombe) [inline src/freenet/client/filter/ContentDataFilter.java:45]: Please add `@Deprecated` to this method so the compiler can tell people not to use this.

File: src/freenet/client/filter/ContentDataFilter.java [hunk 2]
```diff
@@ -35,7 +35,17 @@
...
+			InputStream input, OutputStream output, String charset, HashMap<String, String> otherParams,
```
Comment (Bombe) [inline src/freenet/client/filter/ContentDataFilter.java:46]: Can you use the less restrictive `Map<String, String>` as parameter type here? It shouldn’t break existing code and is clearer to read.

**Review by Bombe — CHANGES_REQUESTED** [review]

File: src/freenet/client/filter/ContentDataFilter.java (hunk 2 above)
Comment (ArneBab) [reply src/freenet/client/filter/ContentDataFilter.java]: that sounds good, yes. And thankfully it’s just a “migrate”-option away to move the implementing classes, too.

File: src/freenet/client/filter/ContentDataFilter.java (hunk 1 above)
Comment (ArneBab) [reply src/freenet/client/filter/ContentDataFilter.java:47]: done :-)

**Comment by ArneBab** [comment]
I asked xor to test the change against WoT — github linked that.

**Comment by ArneBab** [comment]
@Bombe is it good to go?

**Comment by ArneBab** [comment]
WoT is now shown to work.

**Comment by Bombe** [comment]
> @Bombe is it good to go?

Looks good!

**Pull request merged**